### PR TITLE
bump proc-macro-crate versions 

### PR DESCRIPTION
**Pull request by iho**

### CODE REVIEW

File: borsh-derive-internal/src/attribute_helpers.rs [hunk 1]
```diff
@@ -1,34 +1,22 @@
...
-                return true;
-            }
-        }
-    }
-    false
+    attrs
+        .iter()
+        .filter(|attr| attr.path().is_ident("borsh_skip"))
+        .count()
+        > 0
```
Comment (frol) [inline borsh-derive-internal/src/attribute_helpers.rs:8]: https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.any

```suggestion
        .any(|attr| attr.path().is_ident("borsh_skip"))
```

File: borsh-derive/Cargo.toml [hunk 2]
```diff
@@ -18,6 +18,7 @@ proc-macro = true
 [dependencies]
 borsh-derive-internal = { path = "../borsh-derive-internal", version = "0.11.0" }
 borsh-schema-derive-internal = { path = "../borsh-schema-derive-internal", version = "0.11.0" }
-syn = {version = "1", features = ["full", "fold"] }
-proc-macro-crate = "0.1.5"
+syn = { version = "2.0.18", features = ["full", "fold"] }
```
Comment (frol) [inline borsh-derive/Cargo.toml:21]: Unless you explicitly need at least a 2.0.18 version, use `2` instead to let other dependencies prefer some specific version and thus avoid duplication of the syn crate in the dependencies tree.

File: borsh-derive/Cargo.toml [hunk 3]
```diff
@@ -18,6 +18,7 @@ proc-macro = true
...
+proc-macro-crate = "1.3.1"
```
Comment (frol) [inline borsh-derive/Cargo.toml:22]: Same here, consider using `1` as a version.

**Review by frol — CHANGES_REQUESTED** [review]

File: borsh-derive-internal/src/attribute_helpers.rs [hunk 4]
```diff
@@ -1,34 +1,22 @@
...
 }
 
 pub fn contains_initialize_with(attrs: &[Attribute]) -> syn::Result<Option<Path>> {
+    let mut res = None;
```
[the remark this answers is not in the document]
Comment (iho) [reply borsh-derive-internal/src/attribute_helpers.rs:8]: I don't know the way of returning from outer function from closure. It's recommended way of using `parse_nested_meta`: having local variables that closure mutates.

[the remark this answers is not in the document]
Comment (frol) [reply borsh-derive-internal/src/attribute_helpers.rs:8]: I suggest to move this line into the `for` -> `if` block, and return after the `let _ = ...` line.

```rust
for {
    if {
        let res = None;
        let _ = ...;
        return res;
    }
}
None
```

**Review by frol — APPROVED** [review]

**Pull request merged**